core/src/com/unciv/logic/map/TileInfo.kt

```diff
@@ -530,6 +530,8 @@ open class TileInfo {
             resource -> observingCiv != null && hasViewableResource(observingCiv)
             "Water resource" -> isWater && observingCiv != null && hasViewableResource(observingCiv)
             "Natural Wonder" -> naturalWonder != null
+            "Featureless" -> terrainFeatures.isEmpty()
+            "Fresh Water" -> isAdjacentToFreshwater
```

Comment:
Wouldn't this create an infinite loop before? As `isAdjacentToFreshWater` is a lazy that calls `matchesTerrainFilter`?